data-import/desktop-csv-import/desktop-csv-import.adoc

```diff
@@ -0,0 +1,420 @@
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/orders.csv' AS row
+RETURN row
+LIMIT 5
+
+//count data rows in order-details.csv (headers)
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/order-details.csv' AS row
+RETURN row
+LIMIT 8
+----
+
+.Results:
+image:{img}desktop_import_inspect.jpg[link="{img}desktop_import_inspect.jpg",role="popup-link"]
+
+Notice that the orders.csv and the order-details.csv return in a different format from the products.csv.
+This is because those files have headers, so the column names are returned with the values for those rows.
+Since the products.csv does not have column names, then `LOAD CSV` just returns the plain data row from the file.
+
+=== Data types
+
+The `LOAD CSV` command reads all values as a string.
+No matter how the value appears in a file, it will be loaded as a string with `LOAD CSV`.
+So, before we import, we want to ensure we convert any values that are non-string.
+
+There are a variety of conversion functions in Cypher.
+The ones we will use for this exercise are as follows:
+
+* *toInteger():* converts a value to an integer.
+* *toFloat():* converts a value to a float (in this case, for monetary amounts).
+* *datetime():* converts a value to a datetime.
+
+We will look at the values in each CSV file to determine what needs to be converted.
+
+.Products.csv
+The values in the products.csv files are for product ID, product name, and unit cost.
+Product ID looks to be an integer value that increases with each row, so we can convert this to an integer using the `toInteger()` function in Cypher.
+Product name can remain a string because it consists of characters.
+The final column is the product unit cost.
+Though the sample values from our inspection above are all whole numbers, we know that monetary amounts can have decimal place values, as well, so we will convert these values to floats using the `toFloat()` function.
+
+We can see the Cypher to handle all of these conversions below; however, we are still not loading the values into Neo4j yet.
+We are just viewing the CSV files with converted values.
+
+[source, cypher]
+----
+LOAD CSV FROM 'file:///desktop-csv-import/products.csv' AS row
+RETURN toInteger(row[0]), row[1], toFloat(row[2])
+LIMIT 3
+----
+
+image::{img}desktop_import_products_convert.jpg[link="{img}desktop_import_products_convert.jpg",role="popup-link"]
+
+Note that we are using collection positions (row[0], row[1], row[2]) to refer to the columns in the row.
+In a file that has no headers, this is the way to reference values in each position.
+When we look at the next two files that have headers, you will see that we can use the column name to reference the value.
+
+.Orders.csv
+The values in the orders.csv (per the column names) are for orderID, orderDate, and shipCountry.
+Again, we can evaluate the values and determine any conversions to apply.
+
+OrderID looks like it is an integer, so we can convert that using the `toInteger()` function.
```

Comment:
I would be wary with that oftentimes numeric values as IDs might at some point have leading zeros or such.

Reply:
Should we plan for a scenario that doesn't currently exist, though? The numbers increase sequentially and don't resemble generic number ids (more likely chance for leading zeros).